packages/magento-product-configurable/components/ConfigurableProductOptions/ConfigurableProductOption.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  useFormAddProductsToCart,
+  AddProductsToCartMutationVariables,
+} from '@graphcommerce/magento-product/components'
+import {
+  nonNullable,
+  filterNonNullableKeys,
+  SectionHeader,
+  ActionCardListForm,
+  ActionCardItemBase,
+} from '@graphcommerce/next-ui'
+import { useWatch } from '@graphcommerce/react-hook-form'
+import { i18n } from '@lingui/core'
+import { Box, SxProps, Theme } from '@mui/material'
+import { ConfigurableOptionsFragment } from '../../graphql/ConfigurableOptions.gql'
+import { useConfigurableOptionsForSelection } from '../../hooks'
+import {
+  ConfigurableOptionValue,
+  ConfigurableOptionValueFragment,
+} from '../ConfigurableOptionValue'
+
+type Props = NonNullable<
+  NonNullable<ConfigurableOptionsFragment['configurable_options']>[number]
+> & {
+  idx: number
+  index: number
+  optionEndLabels?: Record<string, React.ReactNode>
+  sx?: SxProps<Theme>
+  url_key?: string | null | undefined
+  attribute_code: string
+  render: typeof ConfigurableOptionValue
+}
+
+export function ConfigurableProductOption(props: Props) {
+  const {
+    values,
+    label,
+    index,
+    idx,
+    optionEndLabels,
+    sx,
+    attribute_code,
+    url_key,
+    render,
+    ...other
+  } = props
+  const fieldName = `cartItems.${index}.selected_options.${idx}` as const
+
+  const { control } = useFormAddProductsToCart()
+
+  const selectedOption = useWatch({ control, name: fieldName })
+
+  const selectedOptions = (useWatch({ control, name: `cartItems.${index}.selected_options` }) ?? [])
+    .filter(nonNullable)
+    .filter(Boolean)
+    // This list is sorted by the the order in which the options are displayed.
+    // By slicing were only taking the previous options in consideration and not the options below the current option.
+    // .slice(0, idx)
```

Comment:
Do we want to make this configurable or something? `configurableOptionSelectPrevious`?

If we do not implement this option, we need to be able to unselect certain options because it can happen that

Reply:
For now I've added the ability to deselect chosen options.
Im not sure we really need "configurationOptionsSelectionLogic" ? The user should just be able to configure the product in whichever order he prefers.